linenoise.c

```diff
@@ -879,6 +879,8 @@ linenoiseCompletionCallback * linenoiseSetCompletionCallback(linenoiseCompletion
 
 void linenoiseAddCompletion(linenoiseCompletions *lc, const char *str) {
     lc->cvec = (char **)realloc(lc->cvec,sizeof(char*)*(lc->len+1));
+    if (!lc->cvec)
```

Comment:
Note that that is not the correct way to use realloc because it can introduce a leak. Correct is:

```
char ** x =  (char **)realloc(lc->cvec,sizeof(char*)*(lc->len+1));
if(!x) return;
lc->cvec = x;
...
```

The approach used in this patch will leak if the realloc fails after lc->cvec is allocated.

Reply:
You are right, but this is an existing issue in the code. Will fix it anyways.